tests/greendns_test.py

```diff
@@ -854,3 +857,28 @@ def test_hosts_priority():
 def test_import_rdtypes_then_eventlet():
     # https://github.com/eventlet/eventlet/issues/479
     tests.run_isolated('greendns_import_rdtypes_then_eventlet.py')
+
+
+def test_raise_dns_tcp():
+    # https://github.com/eventlet/eventlet/issues/499
+    try:
+        dns.resolver.query('host.example.com', 'a', tcp=True)
+        assert False, "Should have arisen NXDOMAIN or NoNameservers"
```

Comment:
Test does not ensure that `host.example.com` query to unknown default resolver results in NXDOMAIN.